File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<>();
        for(String c: tokens){
            if(!c.equals("+") && !c.equals("-") && !c.equals("*") && !c.equals("/")){
                stack.push(Integer.parseInt(c));
            }
            else{
                int a = stack.pop();
                int b = stack.pop();
                switch(c){
                    case "+" :
                        stack.push(b+a);
                        break;
                    case "-":
                        stack.push(b-a);
                        break;
                    case "*":
                        stack.push(a*b);
                        break;
                    case "/":
                        if(a != 0) stack.push(b/a);
                        break;
                }
            }
        }
        return stack.peek();
    }
}

File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: list[str]) -> int:
        self.stack = []
        for i in tokens:
            if i != "+" and i != "-" and i != "*" and i != "/":
                self.stack.append(int(i))    
            else:
                a = self.stack.pop()
                b = self.stack.pop()    
                match i:
                    case "+":
                        self.stack.append(b+a)
                    case "-":
                        self.stack.append(b-a)
                    case "*":
                        self.stack.append(b*a)
                    case "/":
                        if a != 0: 
                            self.stack.append(int(b/a))
        return self.stack[-1]